test/integration/tasks.test.js

```diff
@@ -138,6 +138,46 @@ describe("Tasks", function () {
           return done();
         });
     });
+    it("Should get all the list of tasks by paginated", function (done) {
+      const limit = 1;
+      const chaiRequester = chai.request(app).keepOpen();
+      chaiRequester
+        .get("/tasks")
+        .query({ limit })
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.a("object");
+          expect(res.body.tasks).to.be.a("array");
+          expect(res.body.tasks.length).to.equal(limit);
+          const offset = { ...res.body.tasks[0] };
+
+          chaiRequester
+            .get("/tasks")
+            .query({ offset: offset.id })
+            .end((err, res) => {
+              if (err) {
+                return done(err);
+              }
+              expect(res).to.have.status(200);
+              expect(res.body).to.be.a("object");
+              expect(res.body.tasks).to.be.a("array");
+              const taskWithParticipants = res.body.tasks;
+              taskWithParticipants.forEach((task) => {
+                if (task.type === "group") {
+                  expect(task.participants).to.include(appOwner.username);
+                } else {
+                  expect(task.assignee).to.equal(appOwner.username);
+                }
+              });
+              return chaiRequester.close();
+            });
+          return done();
+        });
```

Comment:
Which test checks for the case where no limit is passed, but available tasks are less than the default allowed? What if we encounter an exception that breaks the API?

Reply:
the first test for the "GET /tasks" is the representation that if the limit is not passed the API will not break, but adding to this, I have made changes to the API to better handle the case if there is no limit passed.

Reply:
and the test are still working